### ISSUE TITLE
2-env training also deadlocks after ~3 iterations — world.tick timeout not catching it

### ISSUE THREAD

**Issue by AadityaMishra1** [AadityaMishra1/UrbanZero#4]
## Bug

The `world.tick(seconds=10.0)` fix from issue #3 (commit fa325e1) does NOT prevent the deadlock. Training with **2 envs** (not 4) hangs after iteration 3 (3072 steps) — same silent freeze, no error, no timeout message.

## Key observations

1. Process stays alive (26% CPU, ~1.9GB RAM) but produces no log output
2. All CARLA ports remain reachable via TCP  
3. Both CARLA windows freeze (no physics, no wind — sync mode waiting for tick)
4. **No `world.tick() raised` messages in the log** — the timeout is not firing
5. No NaN-GUARD, reward-guard, or any error/exception in the log
6. The hang occurs AFTER the PPO training pass (iterations 1-3 complete with stats logged), suggesting it happens during the next rollout collection phase
7. This previously worked reliably at 2 envs for 7M steps on the old codebase

## Hypothesis

The deadlock is NOT in `world.tick()` — it's likely in:
- `SubprocVecEnv` worker pipe communication (one worker blocks on `env.step()` and the vec env waits for all workers)
- A sensor `.listen()` callback that never returns
- `apply_batch_sync()` or `apply_control()` blocking inside `step()`
- The new observation gathering code (10-dim state) hitting a CARLA query that hangs

The `world.tick(seconds=10.0)` timeout only covers the tick call itself, not the full `step()` pipeline.

## Suggested fix

Add a **step-level timeout** in `carla_env.py`:
- Wrap the entire `step()` method (or at least the CARLA interaction portion) in a timeout
- If step takes >15s, force-reset the env and return a terminal observation
- Alternatively, use SB3's `VecMonitor` timeout or a custom `TimeoutVecEnv` wrapper

Also consider: does the new code call `world.wait_for_tick()` anywhere? That blocks indefinitely in sync mode.

## Environment

- Branch: `claude/setup-av-training-VetPV` at `e69fa2b`
- 2x CARLA 0.9.15 on Windows (ports 2000, 3000)
- WSL2, RTX 4080 Super
- Prior 7M-step run at 2 envs on OLD codebase worked fine (~100 FPS)

## Complete log (nothing after iteration 3)

```
=== UrbanZero Training ===
  Envs: 2
  Timesteps: 10,000,000

Starting training for 10,000,000 timesteps...
Logging to /home/aadityamishra/urbanzero/logs/v2_rl/PPO_1
---------------------------------
| time/              |          |
|    fps             | 148      |
|    iterations      | 1        |
|    time_elapsed    | 6        |
|    total_timesteps | 1024     |
| train/             |          |
|    ent_coef        | 0.02     |
---------------------------------
------------------------------------------
| time/                   |              |
|    fps                  | 144          |
|    iterations           | 2            |
|    time_elapsed         | 14           |
|    total_timesteps      | 2048         |
| train/                  |              |
|    approx_kl            | 0.0017640111 |
|    clip_fraction        | 0.0203       |
|    clip_range           | 0.2          |
|    ent_coef             | 0.02         |
|    entropy_loss         | -1.84        |
|    explained_variance   | -0.0398      |
|    learning_rate        | 0.0003       |
|    loss                 | 0.004        |
|    n_updates            | 4            |
|    policy_gradient_loss | 0.00082      |
|    std                  | 0.607        |
|    value_loss           | 0.715        |
------------------------------------------
------------------------------------------
| time/                   |              |
|    fps                  | 147          |
|    iterations           | 3            |
|    time_elapsed         | 20           |
|    total_timesteps      | 3072         |
| train/                  |              |
|    approx_kl            | 0.0055222875 |
|    clip_fraction        | 0.0811       |
|    clip_range           | 0.2          |
|    ent_coef             | 0.02         |
|    entropy_loss         | -1.84        |
|    explained_variance   | -0.0421      |
|    learning_rate        | 0.0003       |
|    loss                 | -0.00225     |
|    n_updates            | 8            |
|    policy_gradient_loss | -0.00457     |
|    std                  | 0.607        |
|    value_loss           | 0.628        |
------------------------------------------
[NO FURTHER OUTPUT — process alive but frozen]
```
